marriage_before_conquest.cpp

```diff
@@ -26,8 +26,10 @@ void Mbc::mbc_upper(const vector<point> &p)
         return;
     }
 
-    // this way there is always at lest one point smaller and one point larger then mid_x
-    double mid_x = (p[0].x + p[1].x) / 2;
+    // test cases contain points with integer coordinates
```

Comment:
>     // test cases contain points with integer coordinates
    // we don't want any point to lie on the pivot, that's why adding 0.5
    double mid_x = p[0].x + 0.5;

there are problems when a point lies on the pivot. We can take advantage of the fact that we only have integer coordinates. It is a magic trick but we can turn a blind eye to it.

Reply:
Yes. I had tried to do something to avoid point lies on the pivot, but did not get a general solution. 
If the algorithm goes well, the problem is: if the highest point on the pivot, then many lines cross this point is the bridge. Then, there are the same continuous points in convex_line. We can fix it by unique() the  convex_line.
Are there any other problems with it?